rllib/agents/trainer.py

```diff
@@ -2207,13 +2208,10 @@ def __exit__(self, *args):
 
             def should_stop(self, result):
 
-                # First call to stop, `result` is expected to be None ->
-                # self.failures=0.
+                # First call to stop, `result` is expected to be None.
+                # Return False (meaning: do continue and perform first step).
                 if result is None:
-                    # Fail after n retries.
-                    self.failures += 1
-                    if self.failures > MAX_WORKER_FAILURE_RETRIES:
-                        raise RuntimeError("Failed to recover from worker crash.")
+                    return False
```

Comment:
so I think we should probably have a really long timeout, like if last time I saw good result is 1 hr ago, I will raise Error here, so that then entire Trial errors out, and Tune can restart everything?

Reply:
Or go back to num retries, but make it configurable:
* config.max_worker_failure_retries (respected when either ignore_worker_failure=True or recreate_failed_workers=True); if max_worker_failure_retries=None, don't have a limit here on num attempts.

Lmkwyt ??

Reply:
I like `num_consecutive_worker_failure_tolerance`. Includes the reset idea (after successful iteration) and allows for configuration. Default should be high, like 100.
Will add ...

Reply:
done